packages/contracts-bedrock/src/cannon/MIPS2.sol

```diff
@@ -139,6 +140,7 @@ contract MIPS2 is ISemver {
                 c, m := putField(c, m, 32) // rightThreadStack
                 c, m := putField(c, m, 4) // nextThreadID
             }
+            st.assertExitedIsValid(exited);
```

Comment:
There are a couple of other boolean fields we could validate like this: [thread.exited](https://github.com/ethereum-optimism/optimism/blob/f370113e8de436ad2af5dd093b45366bc8ab352f/packages/contracts-bedrock/src/cannon/MIPS2.sol#L22-L22) and [state.traverseRight](https://github.com/ethereum-optimism/optimism/blob/f370113e8de436ad2af5dd093b45366bc8ab352f/packages/contracts-bedrock/src/cannon/MIPS2.sol#L47).  

I don't think we need to do this extra validation for those?  The `state.exited` field was called out because we do some [integer comparisons with that field](https://github.com/ethereum-optimism/optimism/blob/f370113e8de436ad2af5dd093b45366bc8ab352f/packages/contracts-bedrock/src/cannon/MIPS2.sol#L502-L513).  Open to adding more validation if we think its a good idea though.